smac_planner/src/smac_planner.cpp

```diff
@@ -239,22 +265,32 @@ nav_msgs::msg::Path SmacPlanner::createPlan(
   steady_clock::time_point a = steady_clock::now();
 #endif
 
+  // Choose which costmap to use for the planning
+  std::shared_ptr<nav2_costmap_2d::Costmap2D> costmap;
+  if (shouldDownsample()) {
+    _costmap_downsampler->downsample();
+    _costmap_pub->publishCostmap();
```

Comment:
You should check this publisher if there are any subscribers, that way you don't publish a large costmap when no one's listening.

Reply:
It can actually be a real CPU gain

Reply:
Right, I saw those checks elsewhere as well in the nav2 code, I will add it.